### PR TITLE
Add SNR plot and fix for path issues

### LINKED ISSUE #35: Add SNR plot
The original designer supplies an SNR plot for QC, which we should replicate in Python.

### PR DESCRIPTION
**Pull request by TheJaeger**
This PR introduces SNR graphs as an addition to data quality assurance, to close #35. SNR plots will be saved by default, but can be disabled by parsing the `--noqc` flag. However, this flag also disables all other QC metrics such as those from fitting and eddy. Having a single flag to control all QC metrics is better IMO.

Path checking and making has been overhauled and moved forward to the beginning of the script. Compared to the mess introduced by me when implementing tensor fitting, this is so much better. The checks are simple, and have been extensively tested to work.

Sample SNR plot:
SNR

### CODE REVIEW

File: designer/pydesigner.py [hunk 1]
```diff
@@ -775,4 +801,4 @@
         # reorder tensor for mrtrix3
         DT, KT = img.tensorReorder(img.tensorType())
         dp.writeNii(DT, img.hdr, op.join(metricpath, 'DT'))
-        dp.writeNii(KT, img.hdr, op.join(metricpath, 'KT'))
+        dp.writeNii(KT, img.hdr, op.join(metricpath, 'KT'))
```
Comment (jbteves) [inline designer/pydesigner.py]: Need a newline here

Comment (TheJaeger) [reply designer/pydesigner.py]: What do you mean?

Comment (jbteves) [reply designer/pydesigner.py]: Literally add a line; see that red circle? It's unhappy about it stylistically. It's considered bad formatting.

Comment (jbteves) [reply designer/pydesigner.py]: It's still unhappy, are you doing this on windows? Do you have CRLF set correctly there?

Comment (TheJaeger) [reply designer/pydesigner.py]: I had to open to file in full to see that circle -_-
Surprised to see Git getting upset over the lack of a line.

Comment (jbteves) [reply designer/pydesigner.py]: Of course it is, it's used by the Linux kernel, it adheres to their style guide.

Comment (TheJaeger) [reply designer/pydesigner.py]: TIL. I've made the changes to both files. Apparently it can't have an indentations as well.

Comment (jbteves) [reply designer/pydesigner.py]: Fun fact: that style descends from K&R C, which was first published in... 1978! That style isn't going anywhere.

**Pull request merged**